sdk/main/include/ECDSAsecp256k1PrivateKey.h

```diff
@@ -42,156 +41,185 @@ class ECDSAsecp256k1PrivateKey : public PrivateKey
 {
 public:
   /**
-   * Disallow default construction of an ECDSAsecp256k1PrivateKey, as an uninitialized ECDSAsecp256k1PrivateKey provides
-   * no functionality. Instead, the 'fromString()' or 'fromBytes()' functions should be used.
+   * The number of bytes in an ECDSAsecp256k1PrivateKey.
    */
-  ECDSAsecp256k1PrivateKey() = delete;
+  static constexpr const size_t KEY_SIZE = 32ULL;
+
+  /**
+   * The prefix bytes of a DER-encoded ECDSAsecp256k1PrivateKey.
+   */
+  static inline const std::vector<unsigned char> DER_ENCODED_PREFIX_BYTES = { 0x30, 0x30, 0x02, 0x01, 0x00, 0x30,
+                                                                              0x07, 0x06, 0x05, 0x2B, 0x81, 0x04,
+                                                                              0x00, 0x0A, 0x04, 0x22, 0x04, 0x20 };
+
+  /**
+   * The hex-encoded string of the DER-encoded prefix bytes of an ECDSAsecp256k1PrivateKey.
+   */
+  static inline const std::string DER_ENCODED_PREFIX_HEX = "3030020100300706052B8104000A04220420";
+
+  /**
+   * The maximum number of bytes in an ECDSAsecp256k1PrivateKey signature.
+   */
+  static constexpr const size_t MAX_SIGNATURE_SIZE = 72ULL;
+
+  /**
+   * The number of bytes in a raw ECDSAsecp256k1PrivateKey signature ((r,s) form).
+   */
+  static constexpr const size_t RAW_SIGNATURE_SIZE = 64ULL;
 
   /**
-   * Copy constructor and copy assignment operator can throw OpenSSLException if OpenSSL is unable to serialize the
-   * input key to copy.
+   * The number of bytes in the r value of an ECDSAsecp256k1PrivateKey signature.
    */
-  ~ECDSAsecp256k1PrivateKey() override = default;
-  ECDSAsecp256k1PrivateKey(const ECDSAsecp256k1PrivateKey& other);
-  ECDSAsecp256k1PrivateKey& operator=(const ECDSAsecp256k1PrivateKey& other);
-  ECDSAsecp256k1PrivateKey(ECDSAsecp256k1PrivateKey&& other) noexcept;
-  ECDSAsecp256k1PrivateKey& operator=(ECDSAsecp256k1PrivateKey&& other) noexcept;
+  static constexpr const size_t R_SIZE = 32ULL;
+
+  /**
+   * The number of bytes in the s value of an ECDSAsecp256k1PrivateKey signature.
+   */
+  static constexpr const size_t S_SIZE = 32ULL;
+
+  /**
+   * Disallow default construction of an ECDSAsecp256k1PrivateKey, as an uninitialized ECDSAsecp256k1PrivateKey provides
+   * no functionality. Instead, a factory function should be used.
+   */
+  ECDSAsecp256k1PrivateKey() = delete;
 
   /**
    * Generate a new ECDSAsecp256k1PrivateKey.
    *
    * @return A pointer to the generated ECDSAsecp256k1PrivateKey.
    * @throws OpenSSLException If OpenSSL fails to generate a key.
    */
-  static std::unique_ptr<ECDSAsecp256k1PrivateKey> generatePrivateKey();
+  [[nodiscard]] static std::unique_ptr<ECDSAsecp256k1PrivateKey> generatePrivateKey();
```

Comment:
@alittley curious to your thoughts about having these factory functions return just the value instead of a `std::unique_ptr` (so in this case, just an `ECDSAsecp256k1PrivateKey` as opposed to a `std::unique_ptr<ECDSAsecp256k1PrivateKey>`). I don't necessarily see a reason to lock the user into using exclusively pointers when using these classes.

Reply:
I don't have a strong opinion either way, if you'd prefer returning a value I think that's fine

Reply:
Might have to do a separate effort for this actually, it's a bit more involved than I originally thought